src/components/editor.tsx

```diff
@@ -209,6 +224,60 @@ export default function Editor({ savedata }: Props): React.JSX.Element {
     challenges.forEach((challenge) => changeChallenge(challenge, true));
   };
 
+  const changeCategoryFilter = (
+    category: ChallengeFilterCategory,
+    checked: boolean,
+    activeFilters: ChallengeFilter[],
+    setActiveFilters: React.Dispatch<React.SetStateAction<ChallengeFilter[]>>,
+  ): void => {
+    category.filters.forEach((categoryFilter: ChallengeFilter) =>
+      changeFilter(
+        categoryFilter,
+        checked,
+        activeFilters,
+        setActiveFilters,
+        true,
+      ),
+    );
+    setActiveFilters([...activeFilters]);
```

Comment:
React 18 does automatic state batching, so even if you call `setActiveFilters` multiple times as a result of a user event, only the last call will matter. You can remove the `skipSettingState` argument and the `setActiveFilters([...activeFilters]);` call, and it shouldn't hinder performances (unless this was done for another reason?)

```suggestion
    category.filters.forEach((categoryFilter: ChallengeFilter) =>
      changeFilter(
        categoryFilter,
        checked,
        activeFilters,
        setActiveFilters,
      ),
    );
```